services/agents/azure_database.go

```diff
@@ -0,0 +1,115 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package agents
+
+import (
+	"bytes"
+	"text/template"
+
+	"github.com/percona/pmm/api/agentpb"
+	"github.com/percona/pmm/api/inventorypb"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+const azureDatabaseTemplate = `---
+active_directory_authority_url: "https://login.microsoftonline.com/"
+resource_manager_url: "https://management.azure.com/"
+credentials:
+  client_id: "{{ .ClientID}}"
+  client_secret: "{{ .ClientSecret}}"
+  tenant_id: "{{ .TenantID}}"
+  subscription_id: "{{ .SubscriptionID}}"
+
+targets:
+resource_groups:
+  - resource_group: "{{ .ResourceGroup }}"
+    aggregations:
+      - Average
+{{ .ResourceTypes }}
+    metrics:
+      - name: "cpu_percent"
+      - name: "memory_percent"
+      - name: "storage_percent"
+      - name: "storage_used"
+      - name: "storage_limit"
+      - name: "network_bytes_egress"
+      - name: "network_bytes_ingress"
+`
+
+// azureDatabaseInstance represents credentials informations.
+type azureDatabaseCredentials struct {
+	models.AzureOptions
+	ResourceTypes string
+}
+
+// azureDatabaseExporterConfig returns configuration of azure_database_exporter process.
+//nolint:funlen
+func azureDatabaseExporterConfig(exporter *models.Agent, service *models.Service, redactMode redactMode) (*agentpb.SetStateRequest_AgentProcess, error) {
+	t, err := template.New("credentials").Parse(azureDatabaseTemplate)
+	if err != nil {
+		return nil, err
+	}
+
+	var resourceTypes string
+	switch string(service.ServiceType) {
+	case "mysql":
+		resourceTypes = `    resource_types:
+      - "Microsoft.DBforMySQL/servers"
+      - "Microsoft.DBforMySQL/felexibleServers"`
+	case "maria":
```

Comment:
Does that mean that we added new ServiceType?

Reply:
As far as I see there is no place where we store this value and I would suggest to use new field in AzureOptions to store it.